extensions/amp-ad/0.1/amp-ad-ui.js

```diff
@@ -153,10 +185,42 @@ export class AmpAdUIHandler {
       if (this.state != AdDisplayState.NOT_LAID_OUT) {
         return;
       }
-      this.baseInstance_.togglePlaceholder(true);
+      this.togglePlaceholder_(true);
       this.baseInstance_.toggleFallback(false);
     });
   }
+
+  /**
+   * togglePlaceholder, if use default placeholder, hide it.
+   * @param {boolean} state
+   * @private
+   */
+  togglePlaceholder_(state) {
+    if (this.placeholder_) {
+      this.baseInstance_.togglePlaceholder(state);
+      return;
+    }
+  }
+
+  /**
+   * toggleFallback, if use default fallback, hide it.
+   * @param {boolean} state
+   * @private
+   */
+  toggleFallback_(state) {
+    if (this.fallback_) {
+      this.baseInstance_.toggleFallback(state);
+      return;
+    }
+    if (!this.isExperimentOn_) {
```

Comment:
If we added the "holder" as a fallback, this line and everything after would be unnecessary.